src/sdam/state/mod.rs

```diff
@@ -65,19 +65,20 @@ impl Topology {
             options: options.clone(),
         };
 
-        let state = Arc::new(RwLock::new(TopologyState {
+        let mut topology_state = TopologyState {
             description,
             servers: Default::default(),
-        }));
+        };
 
-        let topology = Topology { state, common };
+        for address in hosts.iter() {
+            topology_state.add_new_server(address.clone(), options.clone());
```

Comment:
I moved the weak reference to the topology off of `Server` and onto `Monitor` to allow us to create the servers before the topology itself was created. That way, we can start the monitors without having to grab the lock here.